### DISCUSSION TITLE
OSError: [Errno 1] EPERM when connecting to WiFi on Pico W

### DISCUSSION THREAD

**Discussion by dumbnerd08** [micropython/micropython#12893] [RP2040 / Pico]
Hello. I am trying to connect to a network on my Pico W. I am using a simple test script:
```py
import network
import time
wlan = network.WLAN(network.STA_IF)
wlan.active(True)
wlan.connect("ssid", "pass")
print(wlan.isconnected())
```
When I run this code, I get the following error:
```
  File "<stdin>", line 5, in <module>
OSError: [Errno 1] EPERM
```
I am using Micropython 1.21.0 from Micropython's website. My editor is Thonny on Debian testing. I have things connected to the board, if that makes a difference.

Interestingly, this worked about an hour ago.

**Comment by davefes** [comment]
At start try:
```py
    sta_if.active(True)

 #  seems more reliable to start with a fresh connect()
    if sta_if.isconnected():
        sta_if.disconnect()
        print ('started in the connected state, but now disconnected')
    else:
        print ('started in the disconnected state')
```
and after a disconnect:
```py
    sta_if.active(False)
```

**Comment by robert-hh** [comment]
After calling wlan.connect() it may take a few seconds before the device is actually connecetd and wlan.isconnected() returns True. See the code example at https://docs.micropython.org/en/latest/wipy/tutorial/wlan.html or here https://docs.micropython.org/en/latest/esp32/quickref.html#networking, which waits in loop until the device is connected.

**Comment by panos-stavrianos** [comment]
Hello, I am having the same problem. I think the issue is on `wlan.active`. 
I tried with `RPI_PICO_W-20241025-v1.24.0.uf2` and some older ones but no luck. 
With older firmwares, I didn't get the log `[CYW43] Failed to start CYW43`, but same behaviour. 
I test it with two `RP2040 PICO W`, unfortunately I get both from the same vendor, so can't dismiss a hardware issue.

```python
from time import sleep
import network

wlan = network.WLAN(network.STA_IF)
sleep(1)
print("WLAN Before Activation:", wlan)
wlan.active(True)  # -> [CYW43] Failed to start CYW43
sleep(1)
print("WLAN After Activation:", wlan)
try:
    wlan.connect("xxxx", "xxxx")  # -> OSError: [Errno 1] EPERM
except Exception as e:
    print("Exception running wlan.connect:", e)
```

```
WLAN Before Activation: <CYW43 STA down 0.0.0.0>
[CYW43] Failed to start CYW43
WLAN After Activation: <CYW43 STA down 0.0.0.0>
Exception running wlan.connect: [Errno 1] EPERM
```

------------------
### UPDATE
Ahh sorry, apparently I got a clone pico, not the original with the CYW43455 chip

**Reply by ikerpf** [reply]
Hi, I think I have the same issue. How did you check that the chip was different from the original CYW43455 and were you able to solve the issue for your chip?

**Reply by panos-stavrianos** [reply]
[This](https://forums.raspberrypi.com/viewtopic.php?t=361532) thread was very helpful, it basically has everything you need, and a lot more. 
In short, mine doesn't have the CYW43455 chip that the native `network` library uses, but it has a ESP8285 chip that you can program with AT commands. 
If yours looks like the picture from the thread, you have it too!

I actually manage to make it work, though it was a bit messy, because you need to flash the ESP8285 first, but following [this](https://forums.raspberrypi.com/viewtopic.php?p=2199304#p2199304) worked well. 
The files you need to download are [here](https://forums.raspberrypi.com/viewtopic.php?p=2169198#p2169198)
Then I use the sample code provided in `TCP_Client.py` (from the files), and connected to the Wi-Fi.
[Here](https://gist.github.com/panos-stavrianos/e8737e0fbe3a1d86b6f616ee94c1a10b) is some junky code from ChatGPT, in an attempt to make a small library. The code works, but barely, so keep that in mind.
